src/app/route/CategoryPage/CategoryPage.component.js

```diff
@@ -142,79 +164,105 @@ class CategoryPage extends Component {
         return path.indexOf('search') === 0 ? null : path;
     }
 
+    /**
+     * Get search parameter for request
+     * @return {String} search request parameter
+     */
+    getSearchParam() {
+        const search = getQueryParam('search', location);
+        return search ? decodeURIComponent(search) : '';
+    }
+
     /**
      * Check if url was changed
      * @return {Boolean}
      */
     urlHasChanged(location, prevProps) {
         const pathnameHasChanged = location.pathname !== prevProps.location.pathname;
-        const searchQueryHasChanged = location.search !== prevProps.location.search;
+        const searchQueryHasChanged = !this.compareQueriesWithoutPage(location.search, prevProps.location.search);
 
         return pathnameHasChanged || searchQueryHasChanged;
     }
 
+    /**
+     * Compares queries with specified filter passed in callback
+     * @returns {Boolean}
+     */
+    compareQueriesWithFilter(currentQuery, previousQuery, callback) {
+        const currentParams = callback(convertQueryStringToKeyValuePairs(currentQuery));
+        const previousParams = callback(convertQueryStringToKeyValuePairs(previousQuery));
+
+        return JSON.stringify(currentParams) === JSON.stringify(previousParams);
+    }
+
+    /**
+     * Compares queries ignoring page number
+     * @return {Boolean}
+     */
+    compareQueriesWithoutPage(currentQuery, previousQuery) {
+        const removePage = (params) => {
+            const { page, ...filteredParams } = params;
+
```

Comment:
No need for enter here ;)

Reply:
looks 'glueish'; however, style guides is tricky question. Maybe we should describe this bit in `eslint` in https://eslint.org/docs/rules/padding-line-between-statements for example, what do you think? Also I have used this one for styles on one of my projects: https://github.com/sds/scss-lint so we could introduce this one to our project as well and then add it to CI for example, I believe by implementing it we could reduce time spent on CR as well. Please let me know your thoughts about it as well.

Reply:
Both should be configured. That is a nice idea!

Reply:
I will move this to separate tickets.

Reply:
Moved into #240

Reply:
Nice, thank you.